src/compiler/types.ts

```diff
@@ -3499,6 +3499,7 @@ namespace ts {
     export interface JsFileExtensionInfo {
         extension: string;
         isMixedContent: boolean;
+        scriptKind?: ScriptKind;

```

Comment:
I think this is populated by passing a `ScriptInfo` as a `JsFileExtensionInfo`.  If that's the case, is this actually optional?  Is the question mark just there in case a consumer has created their own implementation of the interface (i.e. for back compat)?

Reply:
I misunderstood - it's passed by the client.

Reply:
For back-compat reasons for this late fix, I've tried to make the changes as benign as possible, rather than requiring extra fields. I may look for areas when porting to master where changes can be more explicit in changing expectations.